lib/handlers/hard-quote/handler.ts

```diff
@@ -161,61 +174,70 @@ export class QuoteHandler extends APIGLambdaHandler<
   }
 }
 
-export function getCosignerData(request: HardQuoteRequest, quote: QuoteResponse): CosignerData {
-  const decayStartTime = getDecayStartTime(request.tokenInChainId);
-  // default to open order with the original prices
-  let filler = ethers.constants.AddressZero;
-  let inputOverride = BigNumber.from(0);
-  const outputOverrides = request.order.info.outputs.map(() => BigNumber.from(0));
+export function getCosignerData(
+  request: HardQuoteRequest,
+  quote: QuoteResponse,
+  orderType: OrderType
+): CosignerData | V3CosignerData {
+  switch (orderType) {
+    case OrderType.Dutch_V2: {
+      const decayStartTime = getDecayStartTime(request.tokenInChainId);
+      let filler = ethers.constants.AddressZero;
+      let inputOverride = BigNumber.from(0);
+      const outputOverrides = request.order.info.outputs.map(() => BigNumber.from(0));
 
-  // if the quote is better, then increase amounts by the difference
-  if (request.type === TradeType.EXACT_INPUT) {
-    if (quote.amountOut.gt(request.totalOutputAmountStart)) {
-      const increase = quote.amountOut.sub(request.totalOutputAmountStart);
-      // give all the increase to the first (swapper) output
-      outputOverrides[0] = request.order.info.outputs[0].startAmount.add(increase);
-      if (quote.filler) {
-        filler = quote.filler;
+      if (request.type === TradeType.EXACT_INPUT) {
+        if (quote.amountOut.gt(request.totalOutputAmountStart)) {
+          const increase = quote.amountOut.sub(request.totalOutputAmountStart);
+          outputOverrides[0] = request.order.info.outputs[0].startAmount.add(increase);
+          if (quote.filler) {
+            filler = quote.filler;
+          }
+        }
+      } else {
+        if (quote.amountIn.lt(request.totalInputAmountStart)) {
+          inputOverride = quote.amountIn;
+          if (quote.filler) {
+            filler = quote.filler;
+          }
+        }
       }
+
+      const v2Data: CosignerData = {
+        decayStartTime,
+        decayEndTime: getDecayEndTime(request.tokenInChainId, decayStartTime),
+        exclusiveFiller: filler,
+        exclusivityOverrideBps: DEFAULT_EXCLUSIVITY_OVERRIDE_BPS,
+        inputOverride,
+        outputOverrides,
+      };
+      return v2Data;
     }
-  } else {
-    if (quote.amountIn.lt(request.totalInputAmountStart)) {
-      inputOverride = quote.amountIn;
-      if (quote.filler) {
-        filler = quote.filler;
-      }
+
+    case OrderType.Dutch_V3: {
```

Comment:
We actually won't be calling `getCosignerData` with Dutch_V3 since we will have the flag `forceOpenOrder` for now. Instead of this, I'd throw an error here if it's v3 since we're not expecting anyone to use v3 with RFQ yet.